### ISSUE TITLE
squelch recalibration?

### ISSUE THREAD

**Issue by Ruprych** [Tunas1337/UV-K5-Modded-Firmwares#6]
Since "issues" here are also about "new ideas", then I have a question: Do you know, if the squelch calibration values are hidden somewhere in the firmware? Is it possible to recalibrate squelch to higher "service values", or to more widely spaced values, by patching the firmware? Thanks in advance.

**Comment by Tunas1337** [comment]
I believe so. As soon as we figure out what commands are being sent to the chip, then figure out what the squelch commands are, we can experiment with changing those values.

**Comment by Tunas1337** [comment]
For the purpose of keeping you posted on the progress: someone found some reference code for the BK4819 and in there, I found what appears to be code to set the squelch:
![image](https://github.com/Tunas1337/UV-K5-Modded-Firmwares/assets/12097904/8fceab47-4392-4631-ade2-fa86876e33f8)
Now it's just a matter of finding where similar things are done in the firmware and experiment with changing the values.

**Comment by Ruprych** [comment]
GreatThank You very much for the info Wysłane z aplikacji Poczta dla systemu Windows  Od: AndrejWysłano: niedziela, 11 czerwca 2023 10:49Do: Tunas1337/UV-K5-Modded-FirmwaresDW: Ruprych; AuthorTemat: Re: [Tunas1337/UV-K5-Modded-Firmwares] squelch recalibration? (Issue #6) For the purpose of keeping you posted on the progress: someone found some reference code for the BK4819 and in there, I found what appears to be code to set the squelch:Now it's just a matter of finding where similar things are done in the firmware and experiment with changing the values.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

**Comment by Ruprych** [comment]
great
thank you very much for the info

**Comment by marusc1** [comment]
Would be great as in some situations the squelch behaves odd. On some frequencies and antennas it opens at unreasonable high input levels resulting in poor sensitivity with much more sensitivity when opening sguelch by button.

**Comment by Ruprych** [comment]
> Would be great as in some situations the squelch behaves odd. On some frequencies and antennas it opens at unreasonable high input levels resulting in poor sensitivity with much more sensitivity when opening sguelch by button.

Somehow out of subject, but please try to use NFM (narrow) instead of FM (wide) especially on 70 cm band. I have the same issue as described by You, Setting a 4xx MHz channel to NFM helped the radio to properly demodulate it. Most services and repeaters transmit in NFM on 70 cm band and QS poorly tolerates decoding NFM as FM. Squelch issues are a consequence of that.

**Comment by marusc1** [comment]
Good point. This reminds on other issue. It would be great to have separate settings for wide/narrow in receiver and transmitter. Interesting if this can be done as as software mod? The practice on amateur radio is to get highest possible effectiveness and this is done by modulating as much as possible - so much to fill the receivers filter. **To achieve this the receiver has to be set to narrow and the transmitter to wide.**

This is in contrary to adjacent channel emission suppression. But on amateur bands channels with 12,5k spacing are practically not in use at least on simplex channels. So effectiveness is preferred. Just 25k spacing is to be watched.

**Comment by dchauran** [comment]
I haven't tried on 70cm, but on 2m using NFM causes the radio to completely miss signals that are not NFM, and even miss NFM signals that my baofengs can demodulate quite easily. 

As to the squelch, this radio definitely needs help. In AM I have to have my squelch set to 7 or it keys randomly. In FM or FMN, setting my squelch to 1 misses signals that are perfectly intelligible when I have squelch open.

**Comment by Ruprych** [comment]
> I haven't tried on 70cm, but on 2m using NFM causes the radio to completely miss signals that are not NFM, and even miss NFM signals that my baofengs can demodulate quite easily.

heh strange because to me it is the other way around. I have been monitoring transmissions of a public transport service in my hometown for social reasons, and if set to FM the radio missed most of the sound, at the same time signaling a strong Rx, but sound was muted. Since i set the channels to NFM, the radio no longer misses anything, the sound is clear, and the  chatter complete and understandable. And yes, baofengs are of much more tolerance about it. 
Maybe You should set 2 m channels as FM and 70 cm as NFM. Most repeaters in my country work by this rule, particularly FM-Link network. Indeed seems QS is more strict about that than baofengs.

**Comment by MLGHerobrine** [comment]
I'm also having issues with the squelch. At 1 it won't always open for weaker transmissions, and it sometimes even goes in and out in the middle of a "strong" transmission. Would be nice to have better values.
